src/benchmarks/micro/coreclr/GC/Perf_GC.cs

```diff
@@ -0,0 +1,36 @@
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Reflection;
+
+namespace System.Tests
+{
+    [GenericTypeArguments(typeof(byte))]
+    [GenericTypeArguments(typeof(char))]
+    [BenchmarkCategory(Categories.CoreCLR)]
+    public class Perf_GC<T>
+    {
+        [Benchmark]
+        [Arguments(512)]
+        [Arguments(512 * 8)]
+        public T[] NewOperator_Array(int length) => new T[length];
+
+#if !NETFRAMEWORK && !NETCOREAPP2_1 && !NETCOREAPP2_2// API added in .NET Core 3.0 (internal)
+        private readonly Func<int, T[]> _allocateUninitializedArrayDelegate = CreateDelegate(typeof(GC), "AllocateUninitializedArray");
```

Comment:
What's the user facing way of hitting this codepath? I worry this test will be fragile as it depends on non-public members.

Reply:
> I worry this test will be fragile as it depends on non-public members.

Good point!  As of today, this is an internal method used by `ArrayPool` and `StringBuilder`.

@VSadov when do we plan to make it public?

Reply:
We plan to make it public soon(ish). I am not sure what exact shape the API will have. 

We may have additional flags enum, since user may want `uninitialized`, `pinned`, `large` and some combinations are perfectly valid. That may imply some changes in logic - i.e. `pinned` would need to be respected regardless of the allocation size.

Anyways. I think calling it via a delegate is ok for foreseeable future. 
It is the same way we do testing of this API. - same issue since it is not public...